tests/std/tests/P0896R4_ranges_alg_minmax/test.cpp

```diff
@@ -335,6 +337,26 @@ constexpr void mm_constexpr_tests() {
         ProxyRef::no>>();
 }
 
+void test_for_issue1893() {
+    string val           = "meow";
+    string low           = "m";
+    string high          = "n";
+    int projection_count = 0;
+    const auto clamped   = ranges::clamp(
+        ref(val), ref(low), ref(high), [](auto x, auto y) { return x < y; },

```

Comment:
This seems unnecessarily overcomplicated. Why pass `reference_wrapper`s instead of passing the arguments directly and using an identity projection? Why the lambda instead of `ranges::less{}` or `std::less{}`? If we're testing the algorithm beyond just the number of projections, then why don't we verify the result? Why do we verify that exactly one of the arguments is unmodified?

Reply:
Initially, my implementation contained the bug: I wrote `move` instead of the second `forward`.
@stat found the bug. And the test is not for only counting number of how many times the projection was applied but also the test protects us from the wrong implementation (with `std::move` instead `forward`).

I will try a test with your suggestions and see if the test still solves all the problems it should solve.

Reply:
Thanks for the explanation and the comment - it's now clear why the test is doing more than simply counting projection calls.